app/src/forms/email/emailService.js

```diff
@@ -3,176 +3,110 @@ const fs = require('fs');
 const log = require('npmlog');
 const path = require('path');
 const constants = require('../common/constants');
-
 const formService = require('../form/service');
 
-const service = {
+// Helper function used to generate contexts for the ches.merge data argument
+const generateContexts = (type, configData, submission, referer) => {
+  let contextToVal = [];
+  if (type === 'sendStatusAssigned') {
+    contextToVal = [configData.assignmentNotificationEmail];
+  } else if (type === 'sendSubmissionConfirmation' && configData.form.showSubmissionConfirmation) {
+    contextToVal = [configData.body.to];
+  } else if (type === 'sendSubmissionReceived' && configData.form.submissionReceivedEmails) {
+    contextToVal = configData.form.submissionReceivedEmails;
+  }
+  const contexts = [
+    {
+      context: {
+        confirmationNumber: submission.confirmationId,
+        title: configData.title,
+        messageLinkText: configData.messageLinkText,
+        messageLinkUrl: `${service._appUrl(referer)}/form/view?s=${submission.id}`,
+      },
+      to: contextToVal
+    }
+  ];
+  return contexts;
```

Comment:
As this constant is only used once, you can shorthand this by directly returning the array instead (saves a bit of garbage cleanup time):
```js
return [{
  context: {
    confirmationNumber: submission.confirmationId,
    title: configData.title,
    messageLinkText: configData.messageLinkText,
    messageLinkUrl: `${service._appUrl(referer)}/form/view?s=${submission.id}`,
  },
  to: contextToVal
}];
```